ecommerce/customers/models.py

```diff
@@ -67,3 +71,44 @@ def get_short_name(self):
     def email_user(self, subject, message, from_email=None, **kwargs):
         """Send an email to this user."""
         send_mail(subject, message, from_email, [self.email], **kwargs)
+
+
+class Country(BaseAbstractModel):
+    name = models.CharField(max_length=255, verbose_name=_("Country"))
+
+    class Meta:
+        verbose_name = _("county")
+        verbose_name_plural = _("countries")
+
+    def __str__(self):
+        return f"{self.name}"
+
+
+class City(BaseAbstractModel):
+    name = models.CharField(max_length=255, verbose_name=_("City"))
+    country = models.ForeignKey(Country, verbose_name=_("Country"), on_delete=models.PROTECT)
+
+    class Meta:
+        verbose_name = _("city")
+        verbose_name_plural = _("cities")
+
+    def __str__(self):
+        return f"{self.name} - {self.country}"
+
+
+class Address(BaseAbstractModel):
+    name = models.CharField(max_length=255, verbose_name=_("Address Name"))
+    full_name = models.CharField(max_length=255, verbose_name=_("Full Name"))
+    line_1 = models.CharField(max_length=255, verbose_name=_("Line 1"))
+    line_2 = models.CharField(max_length=255, verbose_name=_("Line 2"))
+    phone = models.CharField(max_length=15, verbose_name=_("Phone"), validators=[NUM_REGEX])
+    district = models.CharField(max_length=20, verbose_name=_("District"))
+    postcode = models.CharField(max_length=10, verbose_name=_("Postcode"), validators=[NUM_REGEX])
```

Comment:
maybe you can use different regex for phone and postcode

Reply:
should i modified iban number regex too? cause some countries' iban code also include letters

Reply:
> should i modified iban number regex too? cause some countries' iban code also include letters